packages/sourcecred/src/cli/update.js

```diff
@@ -0,0 +1,224 @@
+import {join as pathJoin} from "path";
+import stringify from "json-stable-stringify";
+import {encode} from "../core/storage/textEncoding";
+
+/**
+  The `update` CLI can be used by instance maintainers to magically update
+  files in their instance after bumping the sourcecred version. This allows us
+  to be more liberal in making breaking changes, as long as we provide updaters
+  in this module to making the transition super easy for instance maintainers.
+
+  To add an updater, create a new function for it below this string and then
+  register your updater by appending it to the `updates` const defined below.
+ */
+
+const v0_9_0 = async () => {
+  const storage = new DiskStorage(process.cwd());
+  const oldDiscordParser = C.object(
+    {
+      guildId: C.string,
+      reactionWeights: C.dict(C.number),
+    },
+    {
+      roleWeightConfig: C.object({
+        defaultWeight: C.number,
+        weights: C.dict(C.number),
+      }),
+      channelWeightConfig: C.object({
+        defaultWeight: C.number,
+        weights: C.dict(C.number),
+      }),
+      propsChannels: C.array(C.string),
+      includeNsfwChannels: C.boolean,
+      defaultReactionWeight: C.number,
+      applyAveragingToReactions: C.boolean,
+    }
+  );
+  const discordConfigPath = pathJoin(
+    "config",
+    "plugins",
+    "sourcecred",
+    "discord",
+    "config.json"
+  );
+  const discordConfig = await loadJsonWithDefault(
+    storage,
+    discordConfigPath,
+    oldDiscordParser,
+    () => null
+  );
+  if (discordConfig) {
+    storage.set(
+      discordConfigPath,
+      encode(
+        stringify(
+          [
+            {
+              guildId: discordConfig.guildId,
+              reactionWeightConfig: {
+                weights: discordConfig.reactionWeights,
+                applyAveraging:
+                  discordConfig.applyAveragingToReactions || false,
+                defaultWeight: discordConfig.defaultReactionWeight || 1,
+              },
+              roleWeightConfig: discordConfig.roleWeightConfig || {
+                defaultWeight: 1,
+                weights: {},
+              },
+              channelWeightConfig: discordConfig.channelWeightConfig || {
+                defaultWeight: 1,
+                weights: {},
+              },
+              propsChannels: discordConfig.propsChannels || [],
+              includeNsfwChannels: discordConfig.includeNsfwChannels || false,
+            },
+          ],
+          {space: 2}
+        )
+      )
+    );
+  }
+};
+
+/**
+  Ascending order registry of updaters of type:
+  [[majorVersion, minorVersion, pathVersion], updaterFunction]
+ */
+const updatesRegistry: $ReadOnlyArray<
+  [[number, number, number], () => Promise<void>]
+> = [[[0, 9, 0], v0_9_0]];
+
+///
+///
+///
+///
+///
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  return 1;
+}
+
+const VERSION_REGEX = /^[0-9]*\.[0-9]*\.[0-9]*$/;
```

Comment:
This limits us to a subset of the [SemVer spec](https://semver.org/#spec-item-9). It'd be great if we could eventually support pre-release extensions and potentially build metadata as well. The latter could just be truncated before lookups.

If we choose not to support any extensions of the SemVer spec right now, our strategy and rationale should probably be laid out in a docstring here, e.g. "we may support pre-release versions as our release process is updated. Presently, we just support `major.minor.release` numbered versions." I think this is defensible as it would be pretty messy to support arbitrarily-named pre-releases.